modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/VirtualFlowTest.java

```diff
@@ -1207,17 +1226,178 @@ public void testSheetChildrenRemainsConstant() {
+            }
+
+            @Override
+            protected double computeMaxHeight(double width) {
+                return computePrefHeight(width);
+            }
+
+        });
+        circleFlow.setCellCount(7);
+        circleFlow.resize(300, 300);
+        circleFlow.layout();
+        circleFlow.layout();
+        return circleFlow;
+    }
+
+    // when moving the flow in one direction, the position of the flow
+    // should not increase in the opposite direction
+    @Test
+    public void testReverseOrder() {
+        double orig = flow.getPosition();
+        flow.scrollPixels(10);
+        double pos = flow.getPosition();
+        assertFalse("Moving in positive direction should not decrease position", pos < orig);
+        flow.scrollPixels(-50);
+        double neg = flow.getPosition();
+        assertFalse("Moving in negative direction should not decrease position", neg > pos);
+    }
+
+    @Test
+    public void testReverseOrderForCircleFlow() {
+        VirtualFlowShim vf = createCircleFlow();
+        double orig = vf.getPosition();
+        vf.scrollPixels(10);
+        double pos = vf.getPosition();
+        assertFalse("Moving in positive direction should not decrease position", pos < orig);
+        vf.scrollPixels(-50);
+        double neg = vf.getPosition();
+        assertFalse("Moving in negative direction should not decrease position", neg > pos);
+    }
+
+    @Test
+    public void testGradualMoveForCircleFlow() {
+        VirtualFlowShim vf = createCircleFlow();
+        vf.resize(600,400);
+        ScrollBar sb = vf.shim_getVbar();
+        double s0 = sb.getLayoutY();
+        double s1 = s0;
+        double position = vf.getPosition();
+        double newPosition =0d;
+        double delta = 0;
+        double newDelta = 0;
+        vf.layout();
+        for (int i = 0; i < 50; i++) {
+            vf.scrollPixels(10);
+            vf.layout();
+            newPosition = vf.getPosition();
+            s1 = sb.getLayoutY();
+            newDelta = newPosition - position;
+            System.err.println("s0 = "+s0+", s1 = "+s1);
+            System.err.println("newDelta = "+newDelta+", delta = "+delta);
+            if (i > 0) {
+                double diff = Math.abs((newDelta-delta)/newDelta);
+                System.err.println("diff = "+diff);
+                // maximum 10% difference allowed
+                assertTrue("Too much variation while scrolling (from "+s0+" to "+s1+")", diff < 0.1);
+            }
+            System.err.println("S1 = "+s1);
+            System.err.println("pos = "+vf.getPosition());
+            assertFalse("Thumb moving in the wrong direction at index ", s1 < s0);
+            s0 = s1;
+            delta = newDelta;
+            position = newPosition;
+        }
+    }
+}
+
+class GraphicalCellStub extends IndexedCellShim<Node> {
+    static ArrayList<Circle> circleList = new ArrayList<>();
+    static {
+        circleList.add(new Circle(10));
```

Comment:
Minor: Could use `List.of`? (just a suggestion)

Reply:
changed